cli/flox/src/commands/environment.rs

```diff
@@ -565,28 +565,57 @@ impl Activate {
         //
         //    eval "$(flox activate)"
         if in_place {
-            Self::activate_non_interactive(
-                &shell,
-                &exports,
-                fixed_up_path_joined,
-                &activation_path,
-            );
+            Self::activate_inplace(&shell, &exports, fixed_up_path_joined, &activation_path);
 
             return Ok(());
         }
 
-        let activate_error =
-            Self::activate_interactive(self.run_args, shell, exports, activation_path, now_active);
+        let activate_error = if !self.run_args.is_empty() {
+            Self::activate_non_interactive(self.run_args, shell, exports, activation_path)
+        } else {
+            Self::activate_interactive(shell, exports, activation_path, now_active)
+        };
         // If we get here, exec failed!
         Err(activate_error)
     }
 
+    fn activate_non_interactive(
+        run_args: Vec<String>,
+        shell: ShellType,
+        exports: HashMap<&str, String>,
+        activation_path: PathBuf,
+    ) -> anyhow::Error {
+        let mut command = Command::new(shell.exe_path());
+        command.envs(exports);
+
+        let script = formatdoc! {"
+                # to avoid infinite recursion sourcing bashrc
```

Comment:
do we need to apply the PATH fixup here as well?

Reply:
Don't we bail if the environment is already active? So I don't think so?

Reply:
i guess that's only relevant for something like

```
eval "$(flox activate -d a)"
tmux
flox activate -d b -- <uses something from a>
```

Reply:
Ah right. I guess that's the same case covered in `activate_inplace` (if a is already active and the user runs `flox activate -d a --in-place`, we'll call `reactivate_in_place`, not `activate_inplace`). But then we're not doing it in `activate_interactive`. I guess we should be consistent between `activate_inplace`, `activate_non_interactive`, and `activate_interactive`? None of those will be called if the environment is already activated. Any of them might be run when `a` has been re-ordered by path helper, and we're activating `b`, right?